notes/src/index.js

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { render } from 'react-dom';
+import { BrowserRouter as Router} from 'react-router-dom';
+import { Route } from 'react-router-dom';
+import { Provider } from 'react-redux';
+import { createStore } from 'redux';
+import noteData from './reducers';
+
+import './index.css';
+import App from './App';
+import NoteViewPage from './components/NoteViewPage';
+import NoteCreatePage from './components/NoteCreatePage';
+import NoteEditPage from './components/NoteEditPage';
+
+let store = createStore(noteData);
+
+render(
+  <Provider store={store}>
+  <Router>
```

Comment:
I noticed you've had to create 3 "wrapper" components just to render the `<Navigation>` component alongside each of the separate components.

Alternatively, you move `<Navigation>` as a "sibling" of `<ListView>`, `<NoteView>`, `<NoteCreate>`, `<NoteEdit>` and get rid of the wrapper components altogether. Like so:

```
<Provider store={store}>
  <Router>
    <Fragment>
      <Navigation />
      <div className="App">
        <Route exact path="/" component={ ListView } />
        <Route exact path="/create" component={ NoteCreate } />
        <Route path="/notes/:number" component={ NoteView } />
        <Route path="/edit/:number" component={ NoteEdit } />
      </div>
    </Fragment>
  </Router>
</Provider>
```

Of course, you'll have to fix the CSS rules so your media query still works and you have to move `<Modal>` from `<NoteViewPage>` to `<NoteView>` but this way simplifies your React structure a bit.

Reply:
Oh that's 1000x better, thank you!